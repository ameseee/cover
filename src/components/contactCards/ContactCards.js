import React, { Component } from 'react';
import PropTypes from 'prop-types';
import CustomMessageForm from '../customMessageForm/CustomMessageForm';

class ContactCards extends Component {
  constructor() {
    super();
    this.state = {
      sendingCustom: false
    }
  }

  openCustomForm = () => {
    this.setState({
      sendingCustom: true
    });
  }

  neverMind = () => {
    this.setState({
      sendingCustom: false
    });
  }

  render() {
    const renderCustomForm =
      this.state.sendingCustom
      ? <CustomMessageForm
        sendCustom={this.props.sendCustom}
        neverMind={this.neverMind}
      />
      : null;

    return (
      <article className="contact-card">
        <h4 className="contact-card-name">{this.props.name}</h4>
        <button
          className="custom-text-btn"
          onClick={() => this.openCustomForm()}>
          Send custom text
        </button>
        <button
          className="location-now-btn"
        >
          Send my location NOW!!
        </button>
          {renderCustomForm}
      </article>
    );
  }

};

ContactCards.propTypes = {
  name: PropTypes.string,
  sendCustom: PropTypes.func,
};

export default ContactCards;
