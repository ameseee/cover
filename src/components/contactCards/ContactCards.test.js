import React from 'react';
import ContactCards from './ContactCards';
import { shallow } from 'enzyme';

describe('<ContactCards />', () => {
  let wrapper;

  beforeEach( () => {
    const wrapper = shallow(<ContactCards />);
  });

  it('should exist', () => {
    expect(wrapper).toBeDefined();
  });

  it.skip('should have a false default state of sendingCustom', () => {
    expect(wrapper.state().sendingCustom).toEqual(false);
  });

  it.skip('should render a contact name', () => {
    const contactName = wrapper.find('h4');

    expect(contactName.length).toEqual(1);
  });

  it.skip('should render a two buttons', () => {
    const buttons = wrapper.find('button');

    expect(buttons.length).toEqual(2);
  });

  it.skip('should change state on input', () => {
    const customBtn = wrapper.find('.custom-text-btn');

    customBtn.simulate('click');
    expect(wrapper.state().sendingCustom).toEqual(true);
  });

});

//should render custom form when state is true
