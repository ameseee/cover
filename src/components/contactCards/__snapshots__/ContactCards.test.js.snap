// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ContactCards /> should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ContactCards />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h4
          className="contact-card-name"
        />,
        <button
          className="custom-text-btn"
          onClick={[Function]}
        >
          Send custom text
        </button>,
        <button
          className="location-now-btn"
          onClick={[Function]}
        >
          Send my location NOW!
        </button>,
        null,
      ],
      "className": "contact-card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "contact-card-name",
        },
        "ref": null,
        "rendered": null,
        "type": "h4",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "Send custom text",
          "className": "custom-text-btn",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Send custom text",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "Send my location NOW!",
          "className": "location-now-btn",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Send my location NOW!",
        "type": "button",
      },
      null,
    ],
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h4
            className="contact-card-name"
          />,
          <button
            className="custom-text-btn"
            onClick={[Function]}
          >
            Send custom text
          </button>,
          <button
            className="location-now-btn"
            onClick={[Function]}
          >
            Send my location NOW!
          </button>,
          null,
        ],
        "className": "contact-card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "contact-card-name",
          },
          "ref": null,
          "rendered": null,
          "type": "h4",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "Send custom text",
            "className": "custom-text-btn",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Send custom text",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "Send my location NOW!",
            "className": "location-now-btn",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Send my location NOW!",
          "type": "button",
        },
        null,
      ],
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
